--- CtpP.nc.old	2008-11-26 00:16:17.000000000 +0100
+++ CtpP.nc	2008-11-26 00:16:25.000000000 +0100
@@ -79,6 +79,9 @@
     interface LinkEstimator;
     interface CtpCongestion;
     interface RootControl;    
+#ifdef CTP_CLEAR
+    interface CtpClear;
+#endif
   }
 
   uses {
@@ -110,6 +113,9 @@
   CollectionPacket = Forwarder;
   CtpPacket = Forwarder;
   CtpCongestion = Forwarder;
+#ifdef CTP_CLEAR
+  CtpClear = Forwarder;
+#endif
   
   components new PoolC(message_t, FORWARD_COUNT) as MessagePoolP;
   components new PoolC(fe_queue_entry_t, FORWARD_COUNT) as QEntryPoolP;
@@ -125,6 +131,9 @@
   components new TimerMilliC() as RoutingBeaconTimer;
   components new TimerMilliC() as RouteUpdateTimer;
   components LinkEstimatorP as Estimator;
+#ifdef CTP_CLEAR
+  CtpClear = Estimator;
+#endif
   Forwarder.LinkEstimator -> Estimator;
 
   components new AMSenderC(AM_CTP_DATA);
@@ -148,6 +157,23 @@
   Forwarder.CtpInfo -> Router;
   Router.CtpCongestion -> Forwarder;
   CtpInfo = Router;
+#ifdef CTP_CLEAR
+  CtpClear = Router;
+#endif
+#if defined(CTP_RADIO_SETTINGS) || defined(LPL_COEXISTENCE)
+  components CC2420ActiveMessageC;
+#endif
+#ifdef CTP_RADIO_SETTINGS
+  components CtpRadioSettingsP;
+  Router.CC2420Packet -> CC2420ActiveMessageC;
+  Router.CtpRadioSettings -> CtpRadioSettingsP;
+  Forwarder.CC2420Packet -> CC2420ActiveMessageC;
+  Forwarder.CtpRadioSettings -> CtpRadioSettingsP;
+#endif
+#ifdef LPL_COEXISTENCE
+  Router.LowPowerListening -> CC2420ActiveMessageC;
+  Forwarder.LowPowerListening -> CC2420ActiveMessageC;
+#endif
 
   components new TimerMilliC() as RetxmitTimer;
   Forwarder.RetxmitTimer -> RetxmitTimer;
