--- CC2420ActiveMessageC.nc.original	2008-07-02 11:53:37.000000000 +0200
+++ CC2420ActiveMessageC.nc	2008-07-02 11:54:31.000000000 +0200
@@ -46,6 +46,7 @@
     interface AMPacket;
     interface Packet;
     interface CC2420Packet;
+    interface CC2420Config;
     interface PacketAcknowledgements;
     interface RadioBackoff[am_id_t amId];
     interface LowPowerListening;
@@ -87,6 +88,7 @@
   PacketLink = LinkC;
   LowPowerListening = LplC;
   CC2420Packet = CC2420PacketC;
+  CC2420Config = CC2420ControlC;
   PacketAcknowledgements = CC2420PacketC;
   
   
